src/librustc_mir/const_eval.rs

```diff
@@ -647,10 +647,14 @@ pub fn const_eval_raw_provider<'a, 'tcx>(
                                                    "could not evaluate static initializer");
             // Ensure that if the above error was either `TooGeneric` or `Reported`
             // an error must be reported.
-            if tcx.sess.err_count() == 0 {
-                tcx.sess.delay_span_bug(err.span,
+            let errs = tcx.sess.track_errors(|| {
```

Comment:
You should put the `err.report_as_error(...)` call above into the closure. We don't need the `err_count` call anymore, it's done by `track_errors`.

The idea was to ensure that calling `report_as_error` actually reports any errors.

You should then be able to yield the values inside the `Err` and `Ok` variants instead of having to carry the `reported_err` variable around

Reply:
I get the point, removed `err_count`.